src/model/element/entity.js

```diff
@@ -177,6 +177,25 @@ class Entity extends Element {
 
     return entity;
   }
+
+  toIndraTemplate(){
+    let name = this.name();
+
+    // if the name is not set it means that there is no info for the entity
+    // so that entity must not be included
+    if ( !name ) {
+      return null;
+    }
+
+    let xref = _.pick(this.association(), ['id', 'namespace']);
```

Comment:
- I believe your intention for `xref` is to retrieve a `dbXref` list if the source is NCBI Gene. This currently always returns `xref=null`.
- the INDRA specific stuff (valid namespaces) could be moved to your api/doc/indra.js module

Reply:
If that is the case is there any need to check for valid namespaces? May I just get the ``dbXref`` even though it is always ``null``currently?

Reply:
I meant that your existing code currently always returns `null` for NCBI Gene records.

Reply:
@jvwong  Okay initially I misunderstood what you said. Actually I did it on purpose because I was thinking that ``hgnc`` is also one of the possible namespaces. Also, since we only care the xrefs for hgnc and chebi I was thinking that for the entries whose namespace is ncbi the xref always should be null here so that the indra query would be formed accordingly.

However, now I suspect that for hgnci entries the namespace is still ncbi but they have some dbxrefs field. And we will consider them if the ``dbXrefs`` is set and the available db is ``hgnc``. Is that correct? If so will we consider the first record from the dbXrefs while forming the query? Is it the same for chebi as well?

Reply:
Yes, we are in agreement. My suggestion below was just to put **everything** (namespace, id; dbXrefs that exist) in an array and let the indra.js module pick out what it can wants. In this way, your function becomes more generic, like  `toSearchTemplates` 

e.g.

```js
[
    {
      "name": "SLC19A1",
      "xrefs": [
        {
          "db": "ncbi",
           "id": "6573"
        },
        ...,
        {
          "db": "HGNC",
           "id": "HGNC:10937"
        } 
      ]
    }
]
```